src/dicg.jl

```diff
@@ -52,6 +52,7 @@ function decomposition_invariant_conditional_gradient(
     lazy=false,
     linesearch_workspace=nothing,
     lazy_tolerance=2.0,
+    num_atoms=100,
```

Comment:
We could also add `extra_vertex_storage` here. This could be, for example, the `pre_computed_set` from Boscia.